Fixes #25197- SearchBar's onChange debounce too slow

Ron-Lavi opened the pull request:
While typing something very quickly and pressing 'Enter' to search,
the onChange function is not fast enough to update the current `searchQuery`.
Removing the debounce on the onChange  fix this problem.

Laviro commented:
[test foreman katello]

Laviro commented:
[test foreman]

Laviro commented:
[test katello]

tbrisker reviewed (changes requested):
Removing the debounce completely makes the search send every single letter i type when typing quickly, which is the original reason the debounce was introduced,

Ron-Lavi commented:
the `onChange` updates the store with the current `searchQuery`,
if it won't update before a fast `Enter` press, the last char/s won't be searched.
I tried to bring the `debounce` to the minimum but it wasn't fast enough if I very(!!) quickly pressed `Enter`.

removing the `debounce` allowed the `onChange` to update the `store` quickly enough to be searched if hitting `Enter` in this speeds.

boaz0 commented:
I agree with @tbrisker. It sounds like the API calls and onChange are coupled, we need to separate those two mechanisms into different functions.
onChange needs to change the Redux state and not to call on API.
The API calls will be done in a different part.

Ron-Lavi commented:
thanks @boaz1337 @tbrisker,
the original `handleSearch` prop was receiving the `searchQuery` from the store, and if you triggered the `handleSearch` in the same second that you changed the input, there was a chance that the `handleSearch` didn't receive the updated `searchQuery`.

I added a debounce to the `handleSearch` and it looks good now.

Laviro commented:
[test katello]

ronlavi2412 committed:
fixes #25197 - onChange debounce too slow

Ron-Lavi commented:
After discussing this with @boaz1337 & @tbrisker offline,
I added the `debounce` to the API call in `AutoCompleteActions`,
and removed all of the unnecessary `debounce` methods in `index`

Ron-Lavi commented:
coverage is reduced only because I am not using `helpers/debounceMethods` anymore.

tbrisker reviewed (approved):
Works well, thanks @Laviro !

tbrisker commented:
:cherries: to 1.20 as fac3d5444
